sway-lib-std/src/identity.sw

```diff
@@ -0,0 +1,10 @@
+//! A wrapper type with 2 variants (`Address` or `ContractId`) sed to represent either in a generic way. The use of this type allows for handling interactions with contracts and addresses in a consistent manner.
+library identity;
```

Comment:
What's the reason for moving `Identity` out of `auth`?

Reply:
I had originally defined `Sender` in the `auth` module because that's where I first needed it.
Now that the usage of `Identity` is going to be more widespread I thought it could stand on its own.
I don't feel strongly about this though; it could really go either way.